common/events_io_test.cpp

```diff
@@ -437,3 +437,42 @@ TEST(EventsIo, DestroyReaderBeforeHandlerIsCalled) {
 
 	EXPECT_TRUE(in_write);
 }
+
+TEST(EventsIo, AsyncIoFromSyncIo) {
+	TestEventLoop loop;
+
+	string input {"abcd"};
+
+	io::ReaderPtr reader = make_shared<io::StringReader>(input);
+
+	vector<uint8_t> output;
+	output.resize(100);
+
+	io::WriterPtr writer = make_shared<io::ByteWriter>(output);
+
+	auto areader = make_shared<events::io::AsyncReaderFromReader>(loop, reader);
```

Comment:
Tbh, I don't think the `From Reader` suffix is necessary, like it's already there in the instantiation?